Added Composer Command global

thomasbilk opened the pull request:
Added a parameter to run composer commands globally. The `working_dir`
parameter is only required if `global_command` is `False`.

Fixes #24052

##### SUMMARY
This PR changes the behaviour of the `composer` command so that global commands can be defined in roles see [composer global](https://getcomposer.org/doc/03-cli.md#global). This also makes the parameter `working_dir` optional.

The reason for this change is that composer provides this functionality and it cannot currently be used with Ansible.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
Composer

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file = /home/user/.ansible.cfg
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION
With this change Ansible can now install packages globally like it is shown in the documentation of composer:

```
composer global require friendsofphp/php-cs-fixer
```

Before that it was impossible to have a command with a white space and no working directory.

thomasbilk committed:
Added Composer Command global

Added a parameter to run composer commands globally. The `working_dir`
parameter is only required if `global_command` is `False`.

Fixes #24052

resmo commented on lib/ansible/modules/packaging/language/composer.py:
```diff
@@ -52,10 +52,19 @@
-            - Directory of your project (see --working-dir).
-        required: true
+            - Directory of your project (see --working-dir). This is required when
+              the command is not run globally.
```
Probably one thing to add is to document (the obvious) that `working_dir` will be ignored if `global_command=True`

~~~diff
+            - Will be ignored if C(global_command=true).
~~~

thomasbilk replied:
I added this to the documentation.

resmo reviewed (changes requested):
Well done, only have a minor improvement request for the doc.

thomasbilk committed:
Added Composer Command global

Added a parameter to run composer commands globally. The `working_dir`
parameter is only required if `global_command` is `False`.

Fixes #24052

resmo commented:
perfect. Thanks!
